mod-clans/src/main/java/gg/solarmc/loader/clans/ClanManager.java

```diff
@@ -0,0 +1,171 @@
+/*
+ *
+ *  * dataloader
+ *  * Copyright © 2021 SolarMC Developers
+ *  *
+ *  * dataloader is free software: you can redistribute it and/or modify
+ *  * it under the terms of the GNU Affero General Public License as
+ *  * published by the Free Software Foundation, either version 3 of the
+ *  * License, or (at your option) any later version.
+ *  *
+ *  * dataloader is distributed in the hope that it will be useful,
+ *  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  * GNU Affero General Public License for more details.
+ *  *
+ *  * You should have received a copy of the GNU Affero General Public License
+ *  * along with dataloader. If not, see <https://www.gnu.org/licenses/>
+ *  * and navigate to version 3 of the GNU Affero General Public License.
+ *
+ */
+
+package gg.solarmc.loader.clans;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import gg.solarmc.loader.Transaction;
+import gg.solarmc.loader.data.DataManager;
+import gg.solarmc.loader.schema.tables.records.ClansClanAlliancesRecord;
+import gg.solarmc.loader.schema.tables.records.ClansClanInfoRecord;
+import org.jooq.DSLContext;
+
+import java.time.Duration;
+import java.util.*;
+
+import static gg.solarmc.loader.schema.tables.ClansClanAlliances.CLANS_CLAN_ALLIANCES;
+import static gg.solarmc.loader.schema.tables.ClansClanInfo.CLANS_CLAN_INFO;
+import static gg.solarmc.loader.schema.tables.ClansClanMembership.CLANS_CLAN_MEMBERSHIP;
+
+
+public class ClanManager implements DataManager {
+
+    private final Cache<Integer,Clan> clans = Caffeine.newBuilder().expireAfterAccess(Duration.ofMinutes(10)).build();
+    private final Cache<Integer,Integer> allianceCache = Caffeine.newBuilder().build();
+
+    /**
+     * Gets a clan from memory cache or from table if not present.
+     * @param transaction the tx
+     * @param id ID of clan requested
+     * @return the clan you asked for idiot
+     * @throws IllegalStateException if the clan isn't present in table
+     */
+    public Clan getClan(Transaction transaction, Integer id) {
+        return clans.get(id, i -> {
+
+            var jooq = transaction.getProperty(DSLContext.class);
+
+            ClansClanInfoRecord rec = jooq.fetchOne(CLANS_CLAN_INFO,CLANS_CLAN_INFO.CLAN_ID.eq(i));
+
+            if (rec == null) {
+                throw new IllegalStateException("No such clan exists!");
+            }
+
+            Set<ClanMember> members = jooq.select(CLANS_CLAN_MEMBERSHIP.USER_ID).from(CLANS_CLAN_MEMBERSHIP)
+                    .where(CLANS_CLAN_MEMBERSHIP.CLAN_ID.eq(id)).fetchSet((rec1) -> new ClanMember(id,rec1.value1(),this));
+
+            ClansClanAlliancesRecord rec1 = jooq.fetchOne(CLANS_CLAN_ALLIANCES,CLANS_CLAN_ALLIANCES.CLAN_ID.eq(id));
```

Comment:
```suggestion
            ClansClanAlliancesRecord recAlly = jooq.fetchOne(CLANS_CLAN_ALLIANCES,CLANS_CLAN_ALLIANCES.CLAN_ID.eq(id));
```
(you might to work a *little* bit on variable names)

Reply:
fixed